src/instructlab/configuration.py

```diff
@@ -260,6 +260,13 @@ class _chat(BaseModel):
     greedy_mode: bool = False
     max_tokens: typing.Optional[int] = None
 
+    @model_validator(mode="before")
+    def fill_defaults(
+        cls, values: typing.Dict[str, typing.Any]
+    ) -> typing.Dict[str, typing.Any]:
+        defaults = {"model": DEFAULTS.DEFAULT_MODEL}
```

Comment:
I probably miss something, but wouldn't this be achieved with pydantic by defining the `model` attribute as:

```
    model: str = Field(default_factory=lambda: "bla")
```

With that, I was able to remove the whole `chat:` section, only leaving `chat: {}`.

To go one step further, I added the default value for `_chat` attribute on `_config` class, and this allowed me to remove the remaining `chat: {}` section.

```
class Config(BaseModel):
    """Configuration for the InstructLab CLI."""

    # required fields
    chat: _chat = _chat()
...
```

What am I missing?

Reply:
PS: I think the proper thing to do for the `_chat` default may require a level of indirection through a `Field` too, like we do for any `@property` based `DEFAULTS.*`, e.g.:

```
chat: _chat = Field(default_factory=_chat)
```

(Note, this was **not** tested.)

Otherwise, test enviroment that overrides base directory for each test case may have these embedded structures unnecessarily cached.